### PR TITLE
[Feature][57836]Get Started Screen

### PR DESCRIPTION
**Pull request by DoQuangKhanh0209**
## Related Tickets
- [https://edu-redmine.sun-asterisk.vn/issues/57826](url)
- [https://edu-redmine.sun-asterisk.vn/issues/57827](url)
- [https://edu-redmine.sun-asterisk.vn/issues/57837](url)
- [https://edu-redmine.sun-asterisk.vn/issues/57838](url)
## WHAT
- Apply MVC Structure
- Create Launch Screen
- Create UI Get Started Screen
- Handle Event Get Started Screen
## Review Checklist

https://user-images.githubusercontent.com/114377716/207565497-3899b256-218e-417b-aa05-68efcdc140af.MP4

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/Zts4P8 |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Message | Sun* commit message | Template: [TASK/FREATURE/BUG/SUPPORT][RedmineID] + message |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: Making-Cocktails/Making-Cocktails/Views/Onboarding/OnboardingCollectionViewCell.swift [hunk 1]
```diff
@@ -0,0 +1,23 @@
+//
+//  OnboardingCollectionViewCell.swift
+//  Making-Cocktails
+//
+//  Created by Quang Khánh on 14/12/2022.
+//
+
+import UIKit
+
+class OnboardingCollectionViewCell: UICollectionViewCell {
```
Comment (thuannv-2723) [inline Making-Cocktails/Making-Cocktails/Views/Onboarding/OnboardingCollectionViewCell.swift]: final

File: Making-Cocktails/Making-Cocktails/Views/Onboarding/OnboardingViewController.swift [hunk 2]
```diff
@@ -0,0 +1,72 @@
+//
+//  OnboardingViewController.swift
+//  Making-Cocktails
+//
+//  Created by Quang Khánh on 14/12/2022.
+//
+
+import UIKit
+
+class OnboardingViewController: UIViewController {
```
Comment (thuannv-2723) [inline Making-Cocktails/Making-Cocktails/Views/Onboarding/OnboardingViewController.swift]: final

File: Making-Cocktails/Making-Cocktails/Views/Onboarding/OnboardingViewController.swift [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+            OnboardingSlide(title: "CONVENIENT", descrition: "One-touch cocktail recipe is yours", image: #imageLiteral(resourceName: "slide2"))
+        ]
+    }
+    
+    @IBAction func nextBtnClick(_ sender: Any) {
+        if currentPage == slides.count - 1 {
+            let controller = storyboard?.instantiateViewController(withIdentifier: "HomeNC") as! UINavigationController
+            controller.modalPresentationStyle = .fullScreen
+            controller.modalTransitionStyle = .flipHorizontal
+            present(controller, animated: true, completion: nil)
```
Comment (thuannv-2723) [inline Making-Cocktails/Making-Cocktails/Views/Onboarding/OnboardingViewController.swift:43]: dublicate

File: Making-Cocktails/Making-Cocktails/Views/Onboarding/OnboardingViewController.swift [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        return CGSize(width: collectionView.frame.width, height: collectionView.frame.height)
+    }
+    
+    func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        let with = scrollView.frame.width
+        currentPage = Int(scrollView.contentOffset.x / with)
+        pageControl.currentPage = currentPage
+    }
+}
```
Comment (thuannv-2723) [inline Making-Cocktails/Making-Cocktails/Views/Onboarding/OnboardingViewController.swift]: break line em nhé

File: Making-Cocktails/Making-Cocktails/Views/Onboarding/OnboardingViewController.swift [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+            present(controller, animated: true, completion: nil)
+        } else {
+            currentPage += 1
+            let indexPath = IndexPath(item: currentPage, section: 0)
+            collectionView.scrollToItem(at: indexPath, at: .centeredHorizontally, animated: true)
+        }
+    }
+}
+
+extension OnboardingViewController: UICollectionViewDelegate, UICollectionViewDataSource, UICollectionViewDelegateFlowLayout {
```
Comment (ghost) [inline Making-Cocktails/Making-Cocktails/Views/Onboarding/OnboardingViewController.swift]: Tách UICollectionViewDataSource ra nha anh

File: Making-Cocktails/Making-Cocktails/Views/Onboarding/OnboardingCollectionViewCell.swift [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    
+    static let identifier = String(describing: OnboardingCollectionViewCell.self)
+    
+    @IBOutlet weak var slideImageView: UIImageView!
```
Comment (ghost) [inline Making-Cocktails/Making-Cocktails/Views/Onboarding/OnboardingCollectionViewCell.swift]: private

File: Making-Cocktails/Making-Cocktails/Views/Onboarding/OnboardingCollectionViewCell.swift [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    @IBOutlet weak var slideTitleLbl: UILabel!
```
Comment (ghost) [inline Making-Cocktails/Making-Cocktails/Views/Onboarding/OnboardingCollectionViewCell.swift]: as above

File: Making-Cocktails/Making-Cocktails/Views/Onboarding/OnboardingViewController.swift [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+        super.viewDidLoad()
+        
+        slides = [
+            OnboardingSlide(title: "WELCOME TO COCKTAIL", descrition: "A world full of recipes", image: #imageLiteral(resourceName: "slide3")),
+            OnboardingSlide(title: "COCKTAIL RECIPES", descrition: "Hundreds of detailed cocktail recipes", image: #imageLiteral(resourceName: "slide1")),
+            OnboardingSlide(title: "CONVENIENT", descrition: "One-touch cocktail recipe is yours", image: #imageLiteral(resourceName: "slide2"))
+        ]
+    }
+    
+    @IBAction func nextBtnClick(_ sender: Any) {
```
Comment (ghost) [inline Making-Cocktails/Making-Cocktails/Views/Onboarding/OnboardingViewController.swift]: private

File: Making-Cocktails/Making-Cocktails/Views/Onboarding/OnboardingViewController.swift [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+        if currentPage == slides.count - 1 {
+            let controller = storyboard?.instantiateViewController(withIdentifier: "HomeNC") as! UINavigationController
+            controller.modalPresentationStyle = .fullScreen
+            controller.modalTransitionStyle = .flipHorizontal
+            present(controller, animated: true, completion: nil)
+        } else {
+            currentPage += 1
+            let indexPath = IndexPath(item: currentPage, section: 0)
+            collectionView.scrollToItem(at: indexPath, at: .centeredHorizontally, animated: true)
```
Comment (ghost) [inline Making-Cocktails/Making-Cocktails/Views/Onboarding/OnboardingViewController.swift]: [.centeredHorizontally, .centeredVertically]

File: Making-Cocktails/Making-Cocktails/Views/Onboarding/OnboardingViewController.swift (hunk 3 above)
Comment (thuannv-2723) [reply Making-Cocktails/Making-Cocktails/Views/Onboarding/OnboardingViewController.swift:43]: Approved

**Review by thuannv-2723 — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**